code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1381,3 +1382,315 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
+		C.loc = src.loc
+		cans -= C
+		if(!cans.len)
+			user << "<span class='warning'> With no remaining containers, the rest of the concoction swirls down the drain...</span>"
+			beaker.reagents.clear_reagents()
+			return
+		if(!beaker.reagents.reagent_list.len)
+			user << "<span class='notice'> The empty container plops out.</span>"
+			beaker.loc = src.loc
+			beaker = null
+			return
+
+/obj/item/weapon/reagent_containers/mortar
+	name = "mortar"
+	desc = "This is a reinforced bowl, used for crushing reagents. Ooga booga Rockstop."
+	icon = 'icons/obj/food.dmi'
+	icon_state = "soup"
+	flags = FPRINT  | OPENCONTAINER
+	volume = 50
+
+	//We want the all-in-one grinder audience
+
+	var/list/blend_items = list (
+		/obj/item/stack/sheet/metal           = list("iron" = 20),
+		/obj/item/stack/sheet/mineral/plasma  = list("plasma" = 20),
+		/obj/item/stack/sheet/mineral/uranium = list("uranium" = 20),
+		/obj/item/stack/sheet/mineral/clown   = list("banana" = 20),
+		/obj/item/stack/sheet/mineral/silver  = list("silver" = 20),
+		/obj/item/stack/sheet/mineral/gold    = list("gold" = 20),
+		/obj/item/weapon/grown/nettle         = list("sacid" = 0),
+		/obj/item/weapon/grown/deathnettle    = list("pacid" = 0),
+		/obj/item/stack/sheet/charcoal        = list("charcoal" = 20),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/soybeans = list("soymilk" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/tomato = list("ketchup" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/corn = list("cornoil" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/wheat = list("flour" = -5),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/ricestalk = list("rice" = -5),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/cherries = list("cherryjelly" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/plastellium = list("plasticide" = 5),
+		/obj/item/seeds = list("blackpepper" = 5),
+		/obj/item/weapon/rocksliver = list("ground_rock" = 30),
+
+		//!Put all recipes above this!
+		/obj/item/weapon/reagent_containers/food = list()
+	)
+
+	var/list/juice_items = list (
+		/obj/item/weapon/reagent_containers/food/snacks/grown/tomato = list("tomatojuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/carrot = list("carrotjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/berries = list("berryjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/banana = list("banana" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/potato = list("potato" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/lemon = list("lemonjuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/orange = list("orangejuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/grown/lime = list("limejuice" = 0),
+		/obj/item/weapon/reagent_containers/food/snacks/watermelonslice = list("watermelonjuice" = 0),

```

Comment:
watermelonslice isn't a /grown/, so what's going to happen when the juicing part of the mortar asks for its potency?

Awful things, probably.